Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -112,61 +112,75 @@ namespace Multiplayer
             {
                 AzNetworking::IConnectionSet& connectionSet = networkInterface->GetConnectionSet();
                 if (connectionSet.GetConnectionCount() > 0)
                 {
-                    connectionSet.VisitConnections([this](AzNetworking::IConnection& connection){ this->DrawConnectionStatus(connection.GetConnectionState()); });
+                    connectionSet.VisitConnections([this](AzNetworking::IConnection& connection)
+                    {
+                        m_hostIpAddress = connection.GetRemoteAddress();
+                        this->DrawConnectionStatus(connection.GetConnectionState(), m_hostIpAddress);
+                    });
                 }
                 else
                 {
                     // If we're a client yet are lacking a connection then we've been unintentionally disconnected
                     // Display a disconnect message in the viewport
-                    DrawConnectionStatus(AzNetworking::ConnectionState::Disconnected);
+                    DrawConnectionStatus(AzNetworking::ConnectionState::Disconnected, m_hostIpAddress);
                 }
             }
         }
     }
 
-    void MultiplayerConnectionViewportMessageSystemComponent::DrawConnectionStatus(AzNetworking::ConnectionState connectionState)
+    void MultiplayerConnectionViewportMessageSystemComponent::DrawConnectionStatus(AzNetworking::ConnectionState connectionState, const AzNetworking::IpAddress& hostIpAddress)
     {
-        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
-        if (!viewport)
-        {
-            return;
-        }
-
-        // Draw the status (example: Connected or Disconnected)
-        const char* connectionStateText = ToString(connectionState).data();
-        const AzFramework::WindowSize viewportSize = viewport->GetViewportSize();
-        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
-        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), static_cast<float>(viewportSize.m_height), 1.0f) + AZ::Vector3(viewportConnectionBottomRightBorderPadding) * viewport->GetDpiScalingFactor();
-        
+        AZ::Color connectionStateColor;
         switch (connectionState)
         {
         case AzNetworking::ConnectionState::Connecting:
-            m_drawParams.m_color = AZ::Colors::Yellow;
+            connectionStateColor = AZ::Colors::Yellow;
             break;
         case AzNetworking::ConnectionState::Connected:
-            m_drawParams.m_color = AZ::Colors::Green;
+            connectionStateColor = AZ::Colors::Green;
             break;
         case AzNetworking::ConnectionState::Disconnecting:
-            m_drawParams.m_color = AZ::Colors::Orange;
+            connectionStateColor = AZ::Colors::Orange;
```

Comment:
nit: Keeping to traffic light system maybe cleaner. Red no connection, yellow == connection state is changing, green == connected.

Reply:
Great idea! Fixed :D